### DISCUSSION TITLE
有沒有辦法能外部註冊 adapters ?

### DISCUSSION THREAD

**Discussion by eric2788** [MountainDash/nonebot-plugin-send-anything-anywhere#94] [Ideas]
我想加 discord adapters, 但是我懶得 pull 下來提 pr，有沒有辦法安裝完插件之後有一個接口能註冊自定義的adapter呢

**Comment by eric2788** [comment]
https://send-anything-anywhere.felinae98.cn/dev/adapter <- 是這個嗎
但是我看到 SupportedAdapters 那邊用的是 onebot

**Comment by felinae98** [comment]
所以你打算做一個 saa 的插件？

**Reply by eric2788** [reply]
也不算，首先是想给自己的bot用，如果没问题到时会再考虑做成插件开放用

**Reply by eric2788** [reply]
我想做成一个让指令和事件所有平台通用的机器人，但是现在的插件和适配器高耦合的问题真的让我傻眼

**Comment by eric2788** [comment]
我稍微看了看和试了试，发现你的 SupportedAdapters 和 SupportedPlatforms 都是 hard code 的, 完全不支持外部注册...
